tests/migration_test.go

```diff
@@ -1266,6 +1266,45 @@ var _ = Describe("[Serial][rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][leve
 			})
 		})
 
+		Context("migration to nonroot", func() {
+
+			BeforeEach(func() {
+				if !checks.HasFeature(virtconfig.NonRoot) {
+					Skip("Test specific to NonRoot featureGate that is not enabled")
+				}
+				tests.DisableFeatureGate(virtconfig.NonRoot)
+			})
+			AfterEach(func() {
+				tests.EnableFeatureGate(virtconfig.NonRoot)
+			})
+
+			It("should migrate root implementation to nonroot", func() {
+				By("Create a VMI that will run root(default)")
+				vmi := tests.NewRandomVMIWithEphemeralDisk(cd.ContainerDiskFor(cd.ContainerDiskAlpine))
```

Comment:
It's good that we can migrate a basic VMI like this from root to non-root. The thing I think i'm still unsure about is if our entire featureset will behave correctly. That's a difficult thing to validate, but we should try to validate some more complex VMs with different types of storage and advanced features.

Reply:
I have added PVC & DV cases. Let me know if you have more test cases in mind.

Reply:
I think I would be mainly concerned about the block migration case, perhaps testing with "[test_id:1853]VM with containerDisk + CloudInit + ServiceAccount + ConfigMap + Secret + DownwardAPI + External Kernel Boot" be enough?

Reply:
@vladikr Added